File: src/java/magmac/app/lang/node/PlantUMLInherits.java
package magmac.app.lang.node;

import magmac.app.compile.node.MapNode;
import magmac.app.compile.node.Node;
import magmac.app.compile.rule.LocatingRule;
import magmac.app.compile.rule.StringRule;
import magmac.app.compile.rule.TypeRule;

public record PlantUMLInherits(String child, String parent) implements PlantUMLRootSegment {
    static TypeRule createInheritsRule() {
        return new TypeRule("inherits", LocatingRule.First(new StringRule("parent"), " <|-- ", new StringRule("child")));
    }

    @Override
    public Node serialize() {
        return new MapNode("inherits")
                .withString("parent", this.parent)
                .withString("child", this.child);
    }
}

File: src/web/magmac/app/lang/node/PlantUMLInherits.ts
import { MapNode } from "../../../../magmac/app/compile/node/MapNode";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { LocatingRule } from "../../../../magmac/app/compile/rule/LocatingRule";
import { StringRule } from "../../../../magmac/app/compile/rule/StringRule";
import { TypeRule } from "../../../../magmac/app/compile/rule/TypeRule";
export class PlantUMLInherits {
	createInheritsRule() : TypeRule {return new TypeRule( "inherits", 0.First( new StringRule( "parent"), " <|-- ", new StringRule( "child")));;}
	serialize() : Node {return new MapNode( "inherits").withString( "parent", 0.parent).withString( "child", 0.child);;}
}
